tests/test.rs

```diff
@@ -243,6 +243,35 @@ pub enum Fizz {
 fn test_enum_unit_variants() {
     let x = Fizz::new_this_is_not_a_drill();
     assert_eq!(x, Fizz::ThisISNotADrill);
+
     let x = Fizz::new_bite_me();
     assert_eq!(x, Fizz::BiteMe);
 }
+
+/// A more involved enum
+#[derive(new, PartialEq, Debug)]
+pub enum Enterprise<T: PartialEq + Debug + Default> {
+    Picard,
```

Comment:
Oh, of course, sorry. I think I read it as a struct variant because of the formatting